### PR TITLE
Save Reward Settings In Metadata

### PR DESCRIPTION
**Pull request by Kennybll**
Fixes Save reward settings (50%/100% and upvote..) in metadata.

Changes:
 - Saves previous post's reward settings under SteemConnect.
 - Then fetched it when creating a new post/draft.

### CODE REVIEW

File: src/client/post/Write/editorActions.js [hunk 1]
```diff
@@ -165,6 +166,8 @@ export function createPost(postData) {
 
     const newBody = isUpdating ? getBodyPatchIfSmaller(postData.originalBody, body) : body;
 
+    saveSettings({ upvoteSetting: upvote, rewardSetting: reward });
```
Comment (Sekhmet) [inline src/client/post/Write/editorActions.js]: This action isn't dispatched so it won't save anything, am I right?

Comment (jm90m) [reply src/client/post/Write/editorActions.js]: yup you're right... it needs to be `dispatch(saveSettings({ upvoteSetting: upvote, rewardSetting: reward })`

File: src/client/settings/settingsReducer.js [hunk 2]
```diff
@@ -7,6 +7,8 @@ const initialState = {
   votePercent: 10000,
   showNSFWPosts: false,
   loading: false,
+  upvoteSetting: true,
+  rewardSetting: '50',
```
Comment (jm90m) [inline src/client/settings/settingsReducer.js]: I see this number '50' a lot throughout this PR, I feel like it should be a constant somewhere, like `export const INITIAL_REWARD_SETTING = '50'` then you can use that constant throughout where ever files need to be done

**Comment by jm90m** [comment]
looking better and better! good work, one thing though instead of having the file named `constants` could you make it more specific and name it to `rewards.js` and move it over to this directory:

`/busy/src/common/constants/`

**Comment by jm90m** [comment]
looks like there are conflicts on here with `src/client/post/Write/editorActions.js` please fix

**Comment by Sekhmet** [comment]
I have just created new post (50% split, no upvote). When I go back to editor it looks like it has been saved properly, but if I refresh page unpvote is checked again.

**Comment by Sekhmet** [comment]
I can't still get that upvote setting to work properly:
upvote

**Comment by jm90m** [comment]
works for me on latest build ?

Where it upvotes itself if checked

busy-test-upvote-pr-12-13

**Comment by Kennybll** [comment]
@jm90m that works, but trying to keep it unchecked is causing me problems. I can't seem to figure it out.

**Comment by jm90m** [comment]
@Kennybll ohhh yeah i see.. gl with it, once you fix it and @Sekhmet approves it, then we can merge.

**Comment by jm90m** [comment]
Just tested this recently, I still can't get this to work for unchecked.

**Comment by Sekhmet** [comment]
I will take a look at this issue.

**Comment by Sekhmet** [comment]
I have updated it there: #1233.

**Pull request closed without merging**